rrmngmnt/common.py

```diff
@@ -37,3 +37,31 @@ def normalize_string(data):
     if isinstance(data, six.text_type):
         data = data.encode('utf-8', errors='replace')
     return data
+
+
+class CommandReader(object):
+
+    def __init__(self, executor, cmd, session_timeout=None, cmd_input=None):
+        self.executor = executor
+        self.cmd = cmd
+        self.session_timeout = session_timeout
+        self.cmd_input = cmd_input
+        self.rc = None
+        self.out = ''
+        self.err = ''
+
+    def read_lines(self):
+        with self.executor.session(self.session_timeout) as ss:
+            command = ss.command(self.cmd)
+            with command.execute() as (in_, out, err):
+                if self.cmd_input:
+                    in_.write(self.cmd_input)
+                    in_.close()
+                while True:
+                    line = out.readline()
+                    self.out += line
+                    if not line:
+                        break
+                    yield line.strip()
+                self.rc = command.rc
+                self.err = err.read()
```

Comment:
I am little bit worried about this practice to read out all error output at the end. it might lead to dead lock in case buffer for err output get exhausted.

The best practice is working working with `select` lib.